bundles/org.openhab.core/src/test/java/org/openhab/core/util/ColorUtilTest.java

```diff
@@ -313,4 +384,241 @@ private void assertRgbEquals(final int[] expected, final int[] actual) {
             assertEquals(expectedS, actualS);
         }
     }
+
+    /**
+     * Test round trips HSB => xyY => HSB
+     */
+    @Test
```

Comment:
These tests should be `@ParametrizedTest`s. You can create the arguments from a method, see https://www.baeldung.com/parameterized-tests-junit-5#6-method

Reply:
Ok. I can change this. My gut suggests there must be a better way of creating the argument stream rather than using nested for loops (as follows). But I cannot think of it. => WDYT?

```
    static class ArgsProviderRGBW implements ArgumentsProvider {
        @Override
        public Stream<? extends Arguments> provideArguments(@Nullable ExtensionContext context) throws Exception {
            List<Arguments> result = new ArrayList<>();
            final double step = 5.0;
            for (double r = 0; r <= 100; r = r + step) {
                for (double g = 0; g <= 100; g = g + step) {
                    for (double b = 0; b <= 100; b = b + step) {
                        for (double w = 0; w <= 100; w = w + step) {
                            result.add(Arguments.of(new double[] { r, g, b, w }));
                        }
                    }
                }
            }
            return result.stream();
        }
    }
```